### DISCUSSION TITLE
[Mac Catalyst] Unable to submit to .pkg file to the App Store due to missing icon

### DISCUSSION THREAD

**Discussion by jbtdevgit** [unoplatform/uno#13342] [Q&A] [answered]
Hi,

I am targeting `.net6.0` and I have been trying to submit a `MacCatalyst` `.pkg` to the app store but an issue is keep on blocking my way `ITMS-90236: Missing required icon - The asset catalog in the application bundle does not contain an icon of size 512x512@2x px. For further assistance, see the Human Interface Guidelines at https://developer.apple.com/design/human-interface-guidelines/foundations/app-icons/. ` The link does not help me. 

I've tried inspecting the generated files in the `publish` folder and I can see the icons are there. 

Here are the things that I've tried but no luck.

- Tried doing this on `Xcode` by creating a simple project, drag and dropped icons on their individual places, and copy pasted the generated `Contents.json` to my Uno.MacCatalyst contents json. The app on xcode builds, and shows the icon.
- Regenerate icons and re-dumped it on the `.appiconset` folder and update necessary `Contents.json`
- Rename icons and repeat what I did on 2nd bullet

Did this for more than 20x already, but I have no luck on getting the app published.
Command used to generate a `.pkg` file `dotnet publish -f net6.0-maccatalyst -c Release`

Here's my `Contents.json`
```
{
	"images": [
		{
			"filename": "icon.png",
			"idiom": "universal",
			"scale": "1x",
			"size": "1024x1024"
		},
		{
			"filename": "icon-16x16~mac.png",
			"idiom": "mac",
			"scale": "1x",
			"size": "16x16"
		},
		{
			"filename": "icon-16x16@2x~mac.png",
			"idiom": "mac",
			"scale": "2x",
			"size": "16x16"
		},
		{
			"filename": "icon-32x32~mac.png",
			"idiom": "mac",
			"scale": "1x",
			"size": "32x32"
		},
		{
			"filename": "icon-32x32@2x~mac.png",
			"idiom": "mac",
			"scale": "2x",
			"size": "32x32"
		},
		{
			"filename": icon-128x128~mac.png",
			"idiom": "mac",
			"scale": "1x",
			"size": "128x128"
		},
		{
			"filename": "icon-128x128@2x~mac.png",
			"idiom": "mac",
			"scale": "2x",
			"size": "128x128"
		},
		{
			"filename": "icon-256x256~mac.png",
			"idiom": "mac",
			"scale": "1x",
			"size": "256x256"
		},
		{
			"filename": "icon-256x256@2x~mac.png",
			"idiom": "mac",
			"scale": "2x",
			"size": "256x256"
		},
		{
			"filename": "icon-512x512~mac.png",
			"idiom": "mac",
			"scale": "1x",
			"size": "512x512"
		},
		{
			"filename": "icon-512x512@2x~mac.png",
			"idiom": "mac",
			"scale": "2x",
			"size": "512x512"
		}
	],
	"info": {
		"version": 1,
		"author": "xcode"
	}
}
```

My `.plist`
```
<key>CFBundleIconName</key>
<string>icon</string>
<key>CFBundleIconFile</key>
<string>AppIcon</string>
```

I also checked the `AppIcon.icns` and it seems like there are only 4 sizes included `16x16, 32x32, 128x128, 256x256` should `512x512 and 1024x1024` be included here as well?
Any ideas on how to fix the issue? Thanks!

**Comment by jeromelaban** [comment]
Thanks for the question. While this is not an issue we've seen before, it feels like [this answer](https://stackoverflow.com/questions/57788242/electron-app-mac-app-store-icon-issue-missing-required-icon) is similar to what you're looking for, though our template should certainly contain the right content from the start.

**Reply by jbtdevgit** [reply]
Hi @jeromelaban ,
Do I just create another folder under the `Resources` along side with the `.appiconset` ?

**Reply by jeromelaban** [reply]
@pictos do you know if we're generating this particular content properly with resizetizer?

**Reply by pictos** [reply]
@jeromelaban I'll double-check on my side and report here asap

**Reply by pictos** [reply]
<img width="969" alt="Screenshot 2023-08-22 at 12 07 04" src="https://github.com/unoplatform/uno/assets/20712372/732e5fd9-802c-4ace-a8fa-648bd61b1351">

Tried on my end, with `dotnet publish -f net7.0-maccatalyst -c Release` and I was able to install the app the appIcon is showing as expected (which, I assume, should be enough to validate that `.appiconset` is being correctly generated by Resizetizer).

But to double check I went to the `.app` file and inspect its contents and here you can see the generated assets.
<img width="917" alt="Screenshot 2023-08-22 at 12 10 49" src="https://github.com/unoplatform/uno/assets/20712372/052235b2-8c9e-4b90-b240-388dabea0224">

<details> 
    <summary>
      Content.json
    </summary>

```json
{
  "images": [
    {
      "idiom": "iphone",
      "size": "20x20",
      "scale": "2x",
      "filename": "iconapp20x20@2x.png"
    },
    {
      "idiom": "ipad",
      "size": "20x20",
      "scale": "2x",
      "filename": "iconapp20x20@2x.png"
    },
    {
      "idiom": "iphone",
      "size": "20x20",
      "scale": "3x",
      "filename": "iconapp20x20@3x.png"
    },
    {
      "idiom": "iphone",
      "size": "29x29",
      "scale": "2x",
      "filename": "iconapp29x29@2x.png"
    },
    {
      "idiom": "ipad",
      "size": "29x29",
      "scale": "2x",
      "filename": "iconapp29x29@2x.png"
    },
    {
      "idiom": "iphone",
      "size": "29x29",
      "scale": "3x",
      "filename": "iconapp29x29@3x.png"
    },
    {
      "idiom": "iphone",
      "size": "40x40",
      "scale": "2x",
      "filename": "iconapp40x40@2x.png"
    },
    {
      "idiom": "ipad",
      "size": "40x40",
      "scale": "2x",
      "filename": "iconapp40x40@2x.png"
    },
    {
      "idiom": "iphone",
      "size": "40x40",
      "scale": "3x",
      "filename": "iconapp40x40@3x.png"
    },
    {
      "idiom": "iphone",
      "size": "60x60",
      "scale": "2x",
      "filename": "iconapp60x60@2x.png"
    },
    {
      "idiom": "iphone",
      "size": "60x60",
      "scale": "3x",
      "filename": "iconapp60x60@3x.png"
    },
    {
      "idiom": "ipad",
      "size": "76x76",
      "scale": "2x",
      "filename": "iconapp76x76@2x.png"
    },
    {
      "idiom": "ipad",
      "size": "83.5x83.5",
      "scale": "2x",
      "filename": "iconapp83.5x83.5@2x.png"
    },
    {
      "idiom": "ios-marketing",
      "size": "1024x1024",
      "scale": "1x",
      "filename": "iconappItunesArtwork.png"
    }
  ],
  "properties": {},
  "info": {
    "version": 1,
    "author": "xcode"
  }
}
```

</details>

**Reply by jbtdevgit** [reply]
Thanks for the responses.
 
I am trying to release to a macOS using a mac-catalyst version, I tried using a .net maui base project with our profiles included and release a mac-catalyst version of that, and I never had any problems -- though the difference is that the `.app` contents of the `maui` is different from what is generated in `uno` so I am not sure how maui generates its assets, so I cannot compare.


@pictos  I also created a new `uno` app using an old template targeting `net6.0` and it seems like both contents are the same as what you have.  Shown below:
<details>
<summary>net6.0 mac-catalyst new app</summary>

```
{
  "images": [
    {
      "scale": "2x",
      "size": "29x29",
      "idiom": "iphone",
      "filename": "Icon58.png"
    },
    {
      "scale": "3x",
      "size": "29x29",
      "idiom": "iphone",
      "filename": "Icon87.png"
    },
    {
      "scale": "2x",
      "size": "40x40",
      "idiom": "iphone",
      "filename": "Icon80.png"
    },
    {
      "scale": "3x",
      "size": "60x60",
      "idiom": "iphone",
      "filename": "Icon180.png"
    },
    {
      "scale": "1x",
      "size": "20x20",
      "idiom": "ipad",
      "filename": "Icon20.png"
    },
    {
      "scale": "2x",
      "size": "20x20",
      "idiom": "ipad",
      "filename": "Icon40.png"
    },
    {
      "scale": "1x",
      "size": "29x29",
      "idiom": "ipad",
      "filename": "Icon29.png"
    },
    {
      "scale": "2x",
      "size": "29x29",
      "idiom": "ipad",
      "filename": "Icon58.png"
    },
    {
      "scale": "1x",
      "size": "40x40",
      "idiom": "ipad",
      "filename": "Icon40.png"
    },
    {
      "scale": "2x",
      "size": "40x40",
      "idiom": "ipad",
      "filename": "Icon80.png"
    },
    {
      "scale": "1x",
      "size": "76x76",
      "idiom": "ipad",
      "filename": "Icon76.png"
    },
    {
      "scale": "2x",
      "size": "20x20",
      "idiom": "iphone",
      "filename": "Icon40.png"
    },
    {
      "scale": "3x",
      "size": "20x20",
      "idiom": "iphone",
      "filename": "Icon60.png"
    },
    {
      "scale": "3x",
      "size": "40x40",
      "idiom": "iphone",
      "filename": "Icon120.png"
    },
    {
      "scale": "2x",
      "size": "60x60",
      "idiom": "iphone",
      "filename": "Icon120.png"
    },
    {
      "scale": "2x",
      "size": "76x76",
      "idiom": "ipad",
      "filename": "Icon152.png"
    },
    {
      "scale": "2x",
      "size": "83.5x83.5",
      "idiom": "ipad",
      "filename": "Icon167.png"
    },
    {
      "scale": "1x",
      "size": "1024x1024",
      "idiom": "ios-marketing",
      "filename": "Icon1024.png"
    }
  ],
  "properties": {},
  "info": {
    "version": 1,
    "author": "xcode"
  }
}
```

</details>
Just a dumb question since I never released a mac-catalyst version of an app. If I am only targeting macOS, Do I need to set both iOS, ipad, and macOS icons when releasing?

And also below are my generated `package.app` contents but it seem s like the `Contents.json` is not included on the on it
![image](https://github.com/unoplatform/uno/assets/44351076/926a23cb-4eb7-4c82-8813-b9f7a8f8b406)

**Reply by jbtdevgit** [reply]
Hi @jeromelaban @pictos ,
I created a new `net6.0` project and created a package using the command `dotnet publish -f net6.0-maccatalyst -c Release` with our current apple profiles and submitted it to the `AppStoreConnect` . Do note that I did not change anything that came along with the base project and only added the needed `plist` `entitlements` and `profiles` in the `mobile.csproj` file. 

I did end up with the same error when submitting it to the AppStore.

If you need the project I can upload it or make a repo for that without the profiles, so you can inspect however you want.
Below is the project screenshot for now, where I used an old uno template that targets `.net6.0` [no gui]
![image](https://github.com/unoplatform/uno/assets/44351076/cbf7a26c-9332-4cc4-9ff4-c04e9ddc0acf)

**Reply by jbtdevgit** [reply]
Hi again,

I did both tests with the new template for both `net6.0` and `net7.0` and below are what I did
- No changes made in the project
- Update `info.plist` and `entitlements.plist` for app store submission only
- Update `mobile.csproj` file for code signing publish using `dotnet publish -f net6.0-maccatalyst -c Release` and `dotnet publish -f net7.0-maccatalyst -c Release`

Findings were:
- `net6.0` has the same problem as the old template, AppStoreConnect still sends me the missing icon
- `net7.0` do not have the problem, and was able to submit a `MacCatalyst` app
- `net6.0` iOS do not have the problem of AOT compilation in comparison with the old `net6.0` template so I was also able to submit the `ipa` file on the AppStoreConnect
- **old** `net6.0` project has an AOT compilation error and unable to create an `ipa` file for `iOS`

Below are the projects used for the submission, do note that this involves submission to the AppStore
[UnoApp3_net7.zip](https://github.com/unoplatform/uno/files/12416610/UnoApp3_net7.zip)
[UnoApp4_net6.zip](https://github.com/unoplatform/uno/files/12416688/UnoApp4_net6.zip)

**Reply by jbtdevgit** [reply, marked as answer]
I have created an issue for this. https://github.com/unoplatform/uno/issues/13427 seems like there are no other workarounds and been stuck with this issue for more than a week already.
